y/y.go

```diff
@@ -267,6 +273,11 @@ func NewThrottle(max int) *Throttle {
 // are already maximum number of workers working. If it detects an error from
 // previously Done workers, it would return it.
 func (t *Throttle) Do() error {
+	t.finLock.Lock()
+	defer t.finLock.Unlock()
+	if t.finished == 1 {
```

Comment:
Why do we need a lock here? Using a `atomic.LoadUint32` should be enough

Reply:
LoadUint32 would just fetch the current value. Consider a scenario where T1 loads the value (false) atomically, then T2 closes the channel, then T1 enters into `for` loop. In that case, we would be sending to the closed channel.

Reply:
Yeah, agreed.